lib/rubocop/cop/sorbet/callback_conditionals_binding.rb

```diff
@@ -50,15 +50,46 @@ def autocorrect(node)
             end
 
             _, _, block = conditional.child_nodes
-            expected_class = node.parent_module_name
+
+            # Find the class node and check if it includes a namespace on the
+            # same line e.g.: Namespace::Class, which will require the fully
+            # qualified name
+
+            klass = node.ancestors.find(&:class_type?)
+
+            expected_class = if klass.children.first.children.first.nil?
+              node.parent_module_name.split("::").last
+            else
+              klass.identifier.source
+            end
 
             bind = if block.begin_type?
               indentation = " " * block.child_nodes.first.loc.column
               "T.bind(self, #{expected_class})\n#{indentation}"
-            elsif block.child_nodes.empty? && !block.ivar_type?
+            elsif block.send_type? && recursive_receiver_is_self?(block)
               "T.bind(self, #{expected_class})."
             else
-              "T.bind(self, #{expected_class}); "
+              do_end_lambda = conditional.source.include?("do") && conditional.source.include?("end")
+
+              unless do_end_lambda
+                # We are converting a one line lambda into a multiline
+                # Remove the space after the `{`
+                if /{\s/.match?(conditional.source)
+                  corrector.remove_preceding(block, 1)
+                end
+
+                # Remove the last space and `}` and re-add it with a line break
```

Comment:
Shouldn't the block be changed in a `do` .. `end` if it's going to be multiline? In our style guide we enforce [BlockDelimiters](https://www.rubydoc.info/gems/rubocop/RuboCop/Cop/Style/BlockDelimiters).

Reply:
Fair 👍